src/main/java/com/godaddy/sonar/ruby/simplecovrcov/SimpleCovRcovJsonParserImpl.java

```diff
@@ -11,49 +16,121 @@
+            reporter.addItem(buildReporterItem(reportItemMapEntry));
+        }
+        return reporter;
+    }
+
+    private ReporterItem buildReporterItem(Map.Entry reporterItemMapEntry) {
+        String filename = reporterItemMapEntry.getKey().toString();
+        JsonArray marksJsonArr = ((JsonArray)reporterItemMapEntry.getValue());
+        Collection<Mark> marks = new ArrayList<>();
+        for(JsonElement marksEl : marksJsonArr) {
+            marks.add(new Mark(marksEl.toString(), marksEl.isJsonNull()));
+        }
+        return new ReporterItem(filename, marks);
+    }
+
+    private Map<String, CoverageMeasuresBuilder> processReporter(Reporter reporter, Map<String, CoverageMeasuresBuilder> coveredFiles) {
+        for (ReporterItem reporterItem : reporter.getItems()) {
+            String filename = reporterItem.getFilename();
+            CoverageMeasuresBuilder fileCoverage = coveredFiles.getOrDefault(filename, CoverageMeasuresBuilder.create());
+            processReporterItem(reporterItem, fileCoverage);
+            coveredFiles.put(reporterItem.getFilename(), fileCoverage);
+        }
+        return coveredFiles;
+    }
+
+    private void processReporterItem(ReporterItem reporterItem, CoverageMeasuresBuilder fileCoverage) {
+        ArrayList<Mark> reporterItemMarks = (ArrayList<Mark>) reporterItem.getMarks();
+        for (int markId = 0; markId < reporterItemMarks.size(); markId++) {
+            Mark mark = reporterItemMarks.get(markId);
+            processMark(mark, markId, fileCoverage);
+        }
+    }
+
+    private void processMark(Mark mark, Integer index, CoverageMeasuresBuilder fileCoverage) {
+        if (mark.getIsNull()) { return; }
+        int hitsCount = mark.getAsLong().intValue();
+        int lineNumber = index + 1;
+        mergeHitsCounters(hitsCount, lineNumber, fileCoverage);
+    }
+
+    /**
+     * Merges hits count of a specified line with already existing coverage results for a given file.
+     *
+     * Coverage information for one file could be presented in several suites. So we have to merge all results per file.
+     * Considering how {@link org.sonar.api.measures.CoverageMeasuresBuilder#setHits} works this a little bit
+     * tricky method is needed: if you already have added some coverage info for the particular line, then you can not
+     * add more information. Please, visit method source code for more understanding. It's very small :-)
+     *
+     * @param hitsCount
+     * @param lineNumber
+     * @param fileCoverage
+     */
+    private void mergeHitsCounters(int hitsCount, int lineNumber, CoverageMeasuresBuilder fileCoverage) {
+        SortedMap<Integer, Integer> hitsByLine = Maps.newTreeMap();
+        hitsByLine.putAll(fileCoverage.getHitsByLine());
+        int oldHits = hitsByLine.getOrDefault(lineNumber, 0);
+        hitsByLine.put(lineNumber, oldHits + hitsCount);
+        fileCoverage.reset();
```

Comment:
Regarding the `CoverageMeasuresBuilder#setHits(int lineId, int hits)`: this is a source code:

```java
if (!hitsByLine.containsKey(lineId)) {
  hitsByLine.put(lineId, hits);
  if (hits > 0) {
    totalCoveredLines += 1;
  }
}
return this;
```

As I understood correctly, if for a given line number the hits information already exists, then if I will call `setHits` for the same line again - nothing will be added. This means, if a coverage information for one file exists in several suites I could not add hits for a line from both, only from the first suite the information will be recorded. That's why I've made so dirty implementation:

1. Save hits by line information inside custom variable
2. Reset hits by line information in the `CoverageMeasuresBuilder` instance
3. Add hits from the suite for a given line 
4. Apply hits by line information in the `CoverageMeasuresBuilder` instance from the variable.

Maybe I've missed something 😄